src/lib/interceptor.ts

```diff
@@ -240,6 +246,54 @@ export class ResponseParsingInterceptor {
   }
 }
 
+export class PreventMutationsInterceptor {
+  private requestInterceptor: grpc.Requester;
+  private blankInterceptor: grpc.Requester;
+
+  constructor() {
+    this.requestInterceptor = this.buildRequester();
+    this.blankInterceptor = buildBlankInterceptor();
+  }
+
+  public intercept(options: grpc.CallOptions, nextCall: NextCall): grpc.InterceptingCall {
+    let isMutationRequest = true;
+
+    if(options?.method_definition?.path) {
+      const { path } = options.method_definition;
+      for(const prefix of NON_MUTABLE_METHOD_PREFIXES) {
```

Comment:
Here you could use lodash _.includes or do an index of within `NON_MUTABLE_METHOD_PREFIXES`.

Reply:
I've avoided using Lodash here as we want to keep this library as dependency free as possible.

Reply:
I'm going to move this whole bit to it's own function as well, that way it can be tested.